source/hud/roverSelector.js

```diff
@@ -0,0 +1,113 @@
+// Copyright 2014 Lockheed Martin Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may 
+// not use this file except in compliance with the License. You may obtain 
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS, 
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+this.draw = function( context, position ) {
+    var rover, icon, posx, posy;
+    for ( var i = 0; i < this.rovers.length; i++ ) {
+        if ( this.rovers[ i ].enabled ) {
+            rover = this.rovers[ i ];
+            icon = this[ rover.id ];
```

Comment:
Everything in the `draw` functions are actually in the hud scope (`this` refers to a hud element in the view, not the node). The method body is copied by the hud driver as well as all of the properties on the hud element nodes. So, the property `rovers` contains a list of objects ([defined here](https://github.com/virtual-world-framework/mars-game/blob/Multi-Rover-HUD/source/hud/roverSelector.js#L75)) that contain an `id` property that is the same as the image name stored on the hud element. The `images` property is not directly copied over to the hud element, rather the individual images belonging to that property are. So, this is just determining if the rover is enabled and then retrieving the icon to display if that's true.

I know it's a little confusing, but it's powerful enough to make it worth it.